### PR TITLE
feat(customers): add rmissing schemas #API-1393

### PR DESCRIPTION
**Pull request by aleksey-a-maltsev**
https://tillhub.atlassian.net/browse/API-1393

### CODE REVIEW

File: lib/v0/branches/read.js [hunk 1]
```diff
@@ -10,23 +10,23 @@ const extended = {
   },
   fa_account_number: {
     description: 'Account number',
-    anyOf: [{
+    oneOf: [{
```
Comment (maurogri-th) [inline lib/v0/branches/read.js:13]: If you were changing it, would had been better to import and use
const { oneOf } = require('../../helpers/payload-or-null')

File: lib/v0/customers/base.js [hunk 2]
```diff
@@ -1,648 +1,518 @@
...
+      items: {
         type: 'object',
-        additionalProperties: true,
+        additionalProperties: false,
         properties: {
-          '1x': {
-            type: 'string',
-            format: 'uri'
+          lines: {
+            oneOf: [
```
Comment (maurogri-th) [inline lib/v0/customers/base.js:232]: ...oneOf({

File: lib/v0/customers/base.js [hunk 3]
```diff
@@ -1,648 +1,518 @@
...
+                  minLength: 1
+                }
+              },
+              {
+                type: 'null'
+              }
+            ]
           },
-          avatar: {
+          street: {
```
Comment (maurogri-th) [inline lib/v0/customers/base.js:247]: use the function

File: lib/v0/customers/base.js [hunk 4]
```diff
@@ -1,648 +1,518 @@
...
-        properties: {
-          email: {
-            type: 'object',
-            additionalProperties: false,
-            properties: {
-              enabled: {
-                type: 'boolean'
+          },
+          street_number: {
+            oneOf: [
```
Comment (maurogri-th) [inline lib/v0/customers/base.js:258]: same

File: lib/v0/customers/base.js [hunk 5]
```diff
@@ -1,648 +1,518 @@
...
+            ]
           },
-          newsletter: {
-            type: 'object',
-            additionalProperties: false,
-            properties: {
-              enabled: {
-                type: 'boolean'
+          locality: {
+            oneOf: [
```
Comment (maurogri-th) [inline lib/v0/customers/base.js:268]: and for the next ones also

File: lib/v0/customers/baseExt.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+const { oneOf } = require('../../helpers/payload-or-null')
+
+module.exports = {
+  insert_id: {
+    description: 'Sequential customer index',
+    type: 'integer'
+  },
+  customer_group: {
+    description: 'The group of customer',
```
Comment (maurogri-th) [inline lib/v0/customers/baseExt.js:9]: is this an UUID?

File: lib/v0/customers/baseExt.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    description: 'Entries customer delegated to',
+    ...oneOf({
+      type: 'array',
+      items: {
+        type: 'string'
+      }
+    })
+  },
+  owner: {
+    description: 'Owner',
```
Comment (maurogri-th) [inline lib/v0/customers/baseExt.js:30]: This description could be a little more specific, its a uuid? reference to which type of entity?

**Review by maurogri-th — CHANGES_REQUESTED** [review]
some comments, nothing really important

File: lib/v0/customers/baseExt.js (hunk 6 above)
Comment (aleksey-a-maltsev) [reply lib/v0/customers/baseExt.js:9]: I can't say. Haven't found any examples of this neither DB nor Dashboard UI. But it returns in response.

**Review by maurogri-th — APPROVED** [review]

**Pull request merged**